### PR TITLE
Fixing The Canons in the ship fight gaining power shot instead on hit.

### PR DESCRIPTION
**Pull request by qwe900**
## Changes Proposed:
-  Remove spelleffekt gain power by hit like melee warri
-  Let the Spellscript handle the power of guns

## SOURCE:
Heres one of hundreds of videos shows he gain power on shot not hit
https://www.youtube.com/watch?v=TCBAUkzgiGg

## Tests Performed:
Compiled in VS2022 Win10 no problems

## How to Test the Changes:

1. .tele icecrowncitadel
2. kill bosses before gunship
3. go to canon. shot

### CODE REVIEW

File: src/server/game/Spells/SpellInfoCorrections.cpp [hunk 1]
```diff
@@ -4458,6 +4458,13 @@ void SpellMgr::LoadSpellInfoCorrections()
             spellInfo->Effects[index].TargetB = 0;
         }
     });
+    // Cannon Blast ICC Horde an Alliance GUn
+
+    ApplySpellFix({ 69399, 70172 }, [](SpellInfo* spellInfo)
+        {
```
Comment (Roxtedy) [inline src/server/game/Spells/SpellInfoCorrections.cpp:4464]: We need to fix space

File: src/server/game/Spells/SpellInfoCorrections.cpp [hunk 2]
```diff
@@ -4459,6 +4459,13 @@ void SpellMgr::LoadSpellInfoCorrections()
         }
     });
 
+     // Cannon Blast ICC Horde an Alliance GUn
```
Comment (heyitsbench) [inline src/server/game/Spells/SpellInfoCorrections.cpp:4462]: GUn

File: src/server/game/Spells/SpellInfoCorrections.cpp [hunk 3]
```diff
@@ -4459,6 +4459,13 @@ void SpellMgr::LoadSpellInfoCorrections()
...
+    ApplySpellFix({ 69399, 70172 }, [](SpellInfo* spellInfo)
+    {
+        //set spellefect power gain of cannon to 0 , to handle it in spellscript
```
Comment (heyitsbench) [inline src/server/game/Spells/SpellInfoCorrections.cpp:4465]: `efect`, `0 ,`

**Comment by Nyeriah** [comment]
Spell effects sometimes are not processed in the order they are expected to be. I don't remember how to circumvent this but the solution presented here doesn't seem correct

**Comment by qwe900** [comment]
lastest change was i mistake, how can i revert this

**Comment by Nyeriah** [comment]
Thanks for your PR, we appreciate your interest in contributing, but unfortunately this solution is a hack and as such cannot be merged as it is

**Pull request closed without merging**